src/Sylius/Behat/Context/Ui/Admin/ManagingPromotionCouponsContext.php

```diff
@@ -0,0 +1,421 @@
+    }
+
+    /**
+     * @When I do not specify its amount
+     */
+    public function iDoNotSpecifyItsAmount()
+    {
+        // Intentionally left blank to fulfill context expectation
+    }
+
+    /**
+     * @When /^I set its usage limit to (\d+)$/
+     */
+    public function iSetItsUsageLimitTo($limit)
+    {
+        $this->createPage->setUsageLimit($limit);
+    }
+
+    /**
+     * @When /^I change its usage limit to (\d+)$/
+     */
+    public function iChangeItsUsageLimitTo($limit)
+    {
+        $this->updatePage->setUsageLimit($limit);
+    }
+
+    /**
+     * @When I specify its amount as :amount
+     */
+    public function iSpecifyItsAmountAs($amount)
+    {
+        $this->generatePage->specifyAmount($amount);
+    }
+
+    /**
+     * @When /^I set its per customer usage limit to (\d+)$/
+     */
+    public function iSetItsPerCustomerUsageLimitTo($limit)
+    {
+        $this->createPage->setCustomerUsageLimit($limit);
+    }
+
+    /**
+     * @When /^I change its per customer usage limit to (\d+)$/
+     */
+    public function iChangeItsPerCustomerUsageLimitTo($limit)
+    {
+        $this->updatePage->setCustomerUsageLimit($limit);
+    }
+
+    /**
+     * @When I make it available till :date
+     */
+    public function iMakeItAvailableTill(\DateTime $date)
+    {
+        $this->createPage->setExpiresAt($date);
+    }
+
+    /**
+     * @When I change expires date to :date
+     */
+    public function iChangeExpiresDateTo(\DateTime $date)
+    {
+        $this->updatePage->setExpiresAt($date);
+    }
+
+    /**
+     * @When I add it
+     * @When I try to add it
+     */
+    public function iAddIt()
+    {
+        $this->createPage->create();
+    }
+
+    /**
+     * @When I save my changes
+     */
+    public function iSaveMyChanges()
+    {
+        $this->updatePage->saveChanges();
+    }
+
+    /**
+     * @When I generate it
+     * @When I try to generate it
+     */
+    public function iGenerateIt()
+    {
+        $this->generatePage->generate();
+    }
+
+    /**
+     * @When /^I delete ("[^"]+" coupon) related to (this promotion)$/
+     * @When /^I try to delete ("[^"]+" coupon) related to (this promotion)$/
+     */
+    public function iDeleteCouponRelatedToThisPromotion(CouponInterface $coupon, PromotionInterface $promotion)
+    {
+        $this->indexPage->open(['promotionId' => $promotion->getId()]);
+        $this->indexPage->deleteResourceOnPage(['code' => $coupon->getCode()]);
+    }
+
+    /**
+     * @Then /^I should see (\d+) coupon on the list related to (this promotion)$/
+     */
+    public function iShouldSeeCouponOnTheList($number, PromotionInterface $promotion)
+    {
+        $this->indexPage->open(['promotionId' => $promotion->getId()]);
+
+        Assert::eq(
+            $number,
+            $this->indexPage->countItems(),
+            sprintf('There should be %s coupons but is %s', $number, $this->indexPage->countItems())
```

Comment:
Skip the `sprintf` and use simple string `'There should be %2$s coupons, but is %s.`.